table/table.go

```diff
@@ -0,0 +1,135 @@
+}
+
+// NewDownloader returns a new Downloader using rhttp.DefaultClient
+func NewDownloader(contentHost string, rendererHost string) Downloader {
+	return NewDownloaderWithClients(rhttp.DefaultClient, contentHost, rhttp.DefaultClient, rendererHost)
+}
+
+// NewDownloaderWithClients returns a new Downloader using the given clients
+func NewDownloaderWithClients(contentClient HTTPClient, contentHost string, rendererClient HTTPClient, rendererHost string) Downloader {
+	return Downloader{
+		contentClient:  contentClient,
+		contentHost:    contentHost,
+		rendererClient: rendererClient,
+		rendererHost:   rendererHost,
+	}
+}
+
+// Type returns the type of file returned by this downloader, a table.
+func (downloader *Downloader) Type() string {
+	return "table"
+}
+
+// QueryParameters returns the format and uri query parameters we require to return a table.
+// 'format' is the format of the file to return - xlsx, csv or html.
+// 'uri' is the location of the file that defines the table (a path that resolves to a .json file in the content server).
+func (downloader *Downloader) QueryParameters() []string {
+	return []string{formatParam, uriParam}
+}
+
+// Download fulfills the Request to download a table.
+func (downloader *Downloader) Download(r *http.Request) (responseBody io.Reader, contentType string, responseStatus int, responseErr error) {
+
+	format := r.URL.Query().Get(formatParam)
+	uri := r.URL.Query().Get(uriParam)
+
+	// call the content server to get the json definition of the table
+	contentRequest, err := createContentRequest(downloader, uri, r)
+	if err != nil {
+		log.ErrorR(r, err, log.Data{"_message": "Unable to create HttpRequest to call content server"})
+		return nil, "", http.StatusInternalServerError, err
+	}
+	contentResponse, err := downloader.contentClient.Do(contentRequest)
+	if err != nil {
+		log.ErrorR(r, err, log.Data{"_message": "Error calling content server", "uri": uri})
+		return nil, "", http.StatusInternalServerError, err
+	}
+	if contentResponse.StatusCode != 200 {
+		err = fmt.Errorf("Unexpected response from content server. Status=%d", contentResponse.StatusCode)
+		log.ErrorR(r, err, log.Data{"uri": uri})
+		return contentResponse.Body, contentResponse.Header.Get("Content-Type"), contentResponse.StatusCode, nil
+	}
+
+	// post the json definition to the renderer
+	renderRequest, err := http.NewRequest("POST", downloader.rendererHost+"/render/"+format, contentResponse.Body)
+	if err != nil {
+		log.ErrorR(r, err, log.Data{"_message": "Unable to create HttpRequest to call render server"})
+		return nil, "", http.StatusInternalServerError, err
+	}
+	copyHeaders(r, renderRequest)
+	renderRequest.Header.Set("Content-Type", "application/json")
+	renderResponse, err := downloader.rendererClient.Do(renderRequest)
+	if err != nil {
+		log.ErrorR(r, err, log.Data{"_message": "Error calling render server", "format": format})
+		return nil, "", http.StatusInternalServerError, err
+	}
```

Comment:
If its the caller who closes the response body can you add this to the godoc stating that its their responsibility to close it - potential for issues if its not closed.